README.md

```diff
@@ -256,6 +264,50 @@ Adding explicit `it` calls works around this error:
  }
 ```
 
+### Common workflow: locking other source sets or configurations
+
+By default, we lock the `main` source set of java projects as production dependencies, and any source set that ends in 
+`test` as test dependencies. This behaviour can be configured per project via the `versionsLock` extension.
+
+For instance, the default behaviour (assuming you only have the default `main` and `test` source sets) can be replicated through:
+
+```gradle
+versionsLock {
+    disableJavaPluginDefaults()  // do not lock any configuration / source set by default
+    production { from sourceSets.main }
+    test { from sourceSets.test }
+}
+``` 
+
+### Common workflow: internal test utility projects
+Sometimes, devs have multiple test projects (unit tests, integration tests) that use a subset of common test classes.
+```
+* :foo
+   \--- source set 'test'
+         \--- :foo-test-common
+* :foo-integration
+   \--- source set 'integrationTest'
+         \--- :foo-test-common
+* :foo-test-common
+   \--- source set 'main'
+```
+
+In this case, we'd like to prevent GCV from locking `:foo-test-common`'s `main` source set to production dependencies,
+and instead treat the entire project as test dependencies. We can do this by disabling the default locked source sets,
+and configuring them ourselves:
+
+```gradle
+# foo-test-common/build.gradle
+apply plugin: 'java'
+
+versionsLock {
+    disableJavaPluginDefaults()  // do not lock any configuration / source set by default
+    test {
+        from sourceSets.main
+    }
```

Comment:
I think we should actually make some little utility for people that says this entire project is a test project.

```gradle
versionsLock {
  testProject()
}
```